### PR TITLE
Revert #411

### PR DESCRIPTION
**Pull request by iGEL**
I completely disagree with #411 and propose to revert it.

The formatting of the number depends on the language in the real world,not on the currency. The best example is the Euro, where it previously enforce the German format on the entire Euro zone. France, Italy and Spain for example write `1 234,56 €`, not `1.234.56 €`. And I think, English speaking countries would expect `1,234.56 €`, even though there is no English speaking country who uses the Euro. In return, we Germans also write `$ 1.234,56`, not `$ 1,234.56`. As I see it, you can't show currencies on a site in German and French now without always passing the params anymore.

The question is, whether the currency should have this information at all. Even though it comes from `money`, which might be used without `i18n`, I think it should be remove for the same reason.

/cc @manuelmeurer, @semmons99, @antstorm

### CODE REVIEW

**Comment by antstorm** [comment]
@iGEL that's a very good point. What you said definitely makes perfect sense. Feels like there should be a separate set of rules for each country, merged with default currency rules.

@manuelmeurer what are your thoughts on this?

**Comment by manuelmeurer** [comment]
I agree, didn't know the Euro (and other currencies?) actually have different formatting in different countries.

**Comment by iGEL** [comment]
@antstorm From my point of view, the money gem doesn't need to know which country uses which decimal rule. Number formatting is not as easy as it might sound, for example Asian countries like China & Korea usually separate by 4 instead of 3 digits, like `1,2345.67` instead of `12,345.67`, see [Chinese_numerals](https://en.wikipedia.org/wiki/Chinese_numerals#Whole_numbers). The money gem could assume the English rules and allow overwriting them globally or per call. It sounds like a I18n concern (the concept, not the gem) and could be handled by a separate gem (like the i18n gem). 

TL;DR: I'd remove that knowledge from the money gem as well.

@manuelmeurer Thanks. I hope I don't sounded to harsh, forgot about the negative bias of written communication. :wink:

**Comment by manuelmeurer** [comment]
@iGEL It's all good. 😄 Actually @semmons99 [had the same concern](https://github.com/RubyMoney/money-rails/pull/411#issuecomment-175771935) early on but I didn't understand what he meant.

**Comment by antstorm** [comment]
@iGEL yeah, that sounds like `money-i18n` or something to me. I remember quite a lot of questions about formatting and it seems like everyone has it's own rules to it.

@manuelmeurer since it was your commit in the first place I want to make sure you're happy with reverting this and there are ways to get similar behaviour with an extra helper method or something like that.

**Comment by manuelmeurer** [comment]
I just checked in my app and setting the locale correctly actually solves my problem mentioned in #411. So I'm good with reverting this! 👍

**Comment by antstorm** [comment]
Great, I'm merging this then!

**Comment by iGEL** [comment]
Thanks, guys!
